xml2rfc/parser.py

```diff
@@ -227,12 +227,10 @@ def getReferenceRequest(self, request, include=False, line_no=0):
                     result = os.path.join(self.source_dir, basename)
                     attempts.append(result)
         else:
-            if self.options and self.options.vocabulary == 'v3':
-                paths = [ request ]
-            elif not request.endswith('.xml'):
-                paths = [ request, request + '.xml' ]
+            if not request.endswith('.xml'):
```

Comment:
On a quick read, I think this is likely to _break_ anyone that happened to be using files locally that didn't have the .xml extension - maybe there aren't any of those, but how would we know?. I think you want to put both request and request with .xml added into the list instead.

Reply:
This is also (or was also) the pure v3 codepath - do we really want to start second-guessing what we're given as v3?

Reply:
> On a quick read, I think this is likely to _break_ anyone that happened to be using files locally that didn't have the .xml extension - maybe there aren't any of those, but how would we know?. I think you want to put both request and request with .xml added into the list instead.

No, it's not going to break for those cases. Because next line adds both variants to the path with and without extension.

Reply:
> This is also (or was also) the pure v3 codepath - do we really want to start second-guessing what we're given as v3?

Original code was not looking at the version attribute rather rely on what's set in command line options to figure out it was v3. I think this doesn't do any harm. Also #1006 adds a warning for use of PIs in v3.

Reply:
(reticent) ok